lib/Operation.php

```diff
@@ -73,7 +75,18 @@ protected function buildCommand(string $template, Node $node, string $event, arr
 
 		if (strpos($command, '%n')) {
 			// Nextcloud relative-path
-			$command = str_replace('%n', escapeshellarg($node->getPath()), $command);
+                        $nodeID = -1;
+                        try {
+                                $nodeID = $node->getId();
+                        } catch (InvalidPathException $e) {
+                        } catch (NotFoundException $e) {
+                        }
+                        $config = new Config('config/');
+                        $base_path = $config->getValue('datadirectory');
+
+                        $path = Filesystem::getLocalFile(Filesystem::getPath($nodeID));
```

Comment:
$nodeID is not necessarily set, since the exceptions are suppressed and not handled further

Reply:
Hmm, what are the cases where $nodeID is not not set and what is the best way to handle the exception? Should we fallback to the old solution (and have a possibly wrong file path on shared folders), or leave the %n path empty?

Reply:
@blizzz